include/dlaf/common/assert.h

```diff
@@ -14,102 +14,100 @@
-/// 1     condition
-/// 2-*   (optional) comma separated part(s) composing the custom message in case of failure
+/// 1. @Expr        (required) : an experssion that returns a bool
+/// 2. @Msg / @Var  (required) : additional message (even if empty) or variable to print
+/// >2. @Msg / @Var (optional) : optional messages / variables to print
 ///
-/// If the condition is false, it will print an error report and call std::terminate()
+/// If @Expr is false, an error message is composed and `std::terminate()` is called. The error message
+/// contains information about @Expr, the origin of the error and any additional messages / variables provided
+/// as arguments. Any variables passed to the argument needs to be printable, i.e. support `operator<<()`.
 ///
-/// The error report will refer to the given origin and can be extended with a custom message composed
-/// concatenating additional parameters given to the macro.
+/// A disabled ASSERT macro has no overhead, @Expr and any additional arguments are not evaluated.
+/// Macros are enabled/disabled as follows:
 ///
-/// If the switch **DLAF_ASSERT_ENABLE** is not defined, this check will not be performed and it will not
-/// add any overhead, nor for the condition evaluation, nor for the message
-#define DLAF_ASSERT_WITH_ORIGIN(origin, ...) DLAF_CHECK_WITH_ORIGIN("", origin, __VA_ARGS__, '\n')
-
-/// **THIS MACRO MUST BE USED WHEN THE CHECK IS NEEDED TO ENSURE A CONDITION THAT HAVE
-/// VERY LOW IMPACT ON PERFORMANCES.**
+///                       Control flag                 Default
+/// DLAF_ASSERT           DLAF_ASSERT_ENABLE           enabled
+/// DLAF_ASSERT_MODERATE  DLAF_ASSERT_MODERATE_ENABLE  enabled in Debug, disabled in Release
+/// DLAF_ASSERT_HEAVY     DLAF_ASSERT_HEAVY_ENABLE     enabled in Debug, disabled in Release
 ///
-/// Parameters:
-/// 1     condition
-/// 2-*   (optional) comma separated part(s) composing the custom message in case of failure
+/// Examples:
 ///
-/// If the condition is false, it will print an error report and call std::terminate()
+/// ```
+/// bool is_sth_true(...) {...}
+/// std::string my_msg(...) {...}
+/// int a = 3;
+/// int b = 4;
 ///
-/// The error report can be extended with a custom message composed concatenating additional parameters
-/// given to the macro.
+/// DLAF_ASSERT(5 == 6, "");                 // ASSERT with no additional information
+/// DLAF_ASSERT(a == b, "my msg", a, b);     // * ASSERT with additional information and argument
+/// DLAF_ASSERT(is_sth_true(), my_msg());    // ASSERT calling functions
+/// ```
 ///
-/// If the switch **DLAF_ASSERT_ENABLE** is not defined, this check will not
-/// be performed and it will not add any overhead, nor for the condition evaluation, nor for the message
-#define DLAF_ASSERT(...) DLAF_ASSERT_WITH_ORIGIN((SOURCE_LOCATION()), __VA_ARGS__)
+/// The output from * is as follows:
+//
+/// ```
+/// [ERROR] <file_name>:<line_number> : <function>
+/// a == b
+/// my msg
+/// a : 3
+/// b : 4
```

Comment:
I think the output is:
```
3
4
```
Am I missing something?

Reply:
No, you are right, I forgot to update the docs, that was what the previous mechanism printed.